app/models/concerns/unparsed_run.rb

```diff
@@ -13,154 +18,286 @@ def parse_into_db
+
+      if wrapper.is_generic_timer && !Run.program_from_attribute(:to_s, run_data[:program]).exchangeable?
+        # We got a file in the exchange format, but the reported timer doesn't support the exchange format.
+        # Wipe the timer to stop from serving the exchange format when a user asks to download the file in the
+        # source timer's format.
+        run_data[:program] = Programs::ExchangeFormat.to_s
+      end
+
+      run_data[:program] = Run.program_from_attribute(:to_s, run_data[:program]).to_sym
+
+      run_data
+    end
+
+    def process_run_metadata(run, run_data)
+      run_data[:game]      = run.game_name
+      run_data[:game_icon] = run.game_icon.presence
+      run_data[:category]  = run.category_name
+      run_data[:name]      = run.extended_name(false)
+
+      run_data[:metadata]                 = {}
+      run_data[:metadata][:srdc_id]       = run.metadata.run_id.presence
+      run_data[:metadata][:platform_name] = run.metadata.platform_name.presence
+      run_data[:metadata][:region_name]   = run.metadata.region_name.presence
+      run_data[:metadata][:uses_emulator] = run.metadata.uses_emulator
+      run_data[:metadata][:variables]     = {}
+
+      run_data[:attempts]  = run.attempt_count
+      run_data[:offset_ms] = run.offset.total_seconds * 1000
+
+      # Grab all run metadata
+      # .with will clean up the iterator when we are done
+      run.metadata.variables.with do |iter|
+        until (variable = iter.next).nil?
+          run_data[:metadata][:variables][variable.name] = variable.value
+        end
+      end
 
-      RunHistory.import(histories.map do |history|
-        RunHistory.new(
-          run_id: id,
+      # Retrieve total playtime, and both sum of bests
+      # .with will clean up any objects that will not be GC'd
+      Parser::LiveSplitCore::Analysis.calculate_total_playtime_for_run(run)
+                                     .with { |analysis| run_data[:total_playtime_ms] = analysis.total_seconds * 1000 }
+      analysis = [
+        Parser::LiveSplitCore::Analysis.calculate_sum_of_best(run, false, false, 0),
+        Parser::LiveSplitCore::Analysis.calculate_sum_of_best(run, false, false, 1)
+      ]
+      run_data[:realtime_sum_of_best_ms] = analysis[0].present? ? analysis[0].total_seconds * 1000 : 0
+      run_data[:gametime_sum_of_best_ms] = analysis[1].present? ? analysis[1].total_seconds * 1000 : 0
+      # See https://github.com/glacials/splits-io/pull/474/files#r241242051
+      analysis.compact.each(&:dispose)
+
+      # Set run duration
+      last_seg = run.segment(run.len - 1)
+      run_data[:realtime_duration_ms] = last_seg.personal_best_split_time.real_time&.total_seconds&.*(1000) || 0
+      run_data[:gametime_duration_ms] = last_seg.personal_best_split_time.game_time&.total_seconds&.*(1000) || 0
+
+      run_data[:default_timing] = Run::REAL
+      if run_data[:realtime_duration_ms].zero? && run_data[:gametime_duration_ms].positive?
```

Comment:
Unrelated to this branch, but we might want to convert this with `zero_to_nil` and check for `nil?` instead of `zero?`. Not sure if it will make any difference in behavior as I'm not sure if this value can ever be `nil` before we touch it, but it would be more consistent with the code above now that times (at least on our end) can be `nil`.

cc @jamacanbacn who noticed some runs with default timings of `realtime` when it felt like it should be `gametime`, perhaps this could be the cause.

Reply:
Sounds good for the future.  @jamacanbacn if you have a link to those runs that were being classified incorrectly that would be nice to find the root cause.